tests/unit/test_meta_data.py

```diff
@@ -51,14 +74,17 @@ def test_sensor_meta_data():
 
 
 def test_dict_like_access():
-    assert power_grid_meta_data["input"]["node"].dtype == power_grid_meta_data["input"]["node"]["dtype"]
+    assert (
+        power_grid_meta_data[DatasetType.input][ComponentType.node].dtype
+        == power_grid_meta_data[DatasetType.input][ComponentType.node]["dtype"]
+    )
 
 
 def test_all_datasets():
     assert set(power_grid_meta_data.keys()) == {
-        "input",
-        "update",
-        "sym_output",
-        "asym_output",
-        "sc_output",
+        DatasetType.input,
+        DatasetType.update,
+        DatasetType.sym_output,
+        DatasetType.asym_output,
+        DatasetType.sc_output,
```

Comment:
we need both the `str` and the `DatasetType` tests here, because it involves a backwards-compatibility check
```suggestion
    assert set(power_grid_meta_data.keys()) == {
        "input",
        "update",
        "sym_output",
        "asym_output",
        "sc_output",
        DatasetType.input,
        DatasetType.update,
        DatasetType.sym_output,
        DatasetType.asym_output,
        DatasetType.sc_output,
```